libs/vectorized-exec-spi/src/main/java/org/opensearch/vectorized/execution/spi/RecordBatchStream.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.vectorized.execution.spi;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Represents a stream of record batches from a DataFusion query execution.
+ * This interface provides access to query results in a streaming fashion.
+ */
+public interface RecordBatchStream extends AutoCloseable {
```

Comment:
Maybe this can be generic typed `RecordBatchStream<T>`

Reply:
Yes makes sense. I had a todo in CSV RBS to refactor a bit